drivers/adc/ad4858/ad4858.c

```diff
@@ -0,0 +1,1130 @@
+/***************************************************************************//**
+ *   @file   ad4858.c
+ *   @brief  Source file for the ad4858 drivers
+********************************************************************************
+ * Copyright 2023(c) Analog Devices, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include "ad4858.h"
+#include "ad4858_regs.h"
+#include "no_os_delay.h"
+#include "no_os_print_log.h"
+
+/**
+ * @brief Check big endianess of the host processor.
+ * @return Big endianess status (true/false)
+ */
+static bool _is_big_endian(void)
```

Comment:
check please if you can't use already some functions available in https://github.com/analogdevicesinc/no-OS/blob/master/util/no_os_util.c or https://github.com/analogdevicesinc/no-OS/blob/master/util/

Otherwise add these generic functions (_is_big_endian and _memswap64) there in a separate commit prior to the driver addition.

Reply:
I didn't find the function in util.c that does what I intended- basically wanted to swap the memory content at byte level and also depending upon the step level (e.g. how many bytes I want to swap). Also no function to check the endianness as well. Just to clarify, I have taken these functions from existing ad4170 drivers (developed by Darius) which is present there on our internal bitbucket repo.
Just to be clear, do you suggest to add two separate commits for this? first commit with only these two functions and 2nd with other driver functions?

Reply:
sorry i wasn't explicit enough. the functions for memswap and is_big_endian should be placed in the no_os_util.c file prior to the driver addition. these are generic functions that should be implemented only once and should not be dependent on a specific part driver.